pkg/apps/ntp.go

```diff
@@ -0,0 +1,73 @@
+package apps
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/flanksource/konfigadm/pkg/phases"
+	. "github.com/flanksource/konfigadm/pkg/types"
+	"github.com/spf13/afero"
+)
+
+var ntpPhase ntp = ntp{
+	serviceVerifier: phases.DefaultServiceVerifier{},
+}
+
+// NTP represents a phase which gives information
+// on how to install NTP on the system
+var NTP Phase = ntpPhase
+
+type ntp struct {
+	serviceVerifier ServiceVerifier
+}
+
+func (n ntp) ApplyPhase(sys *Config, ctx *SystemContext) ([]Command, Filesystem, error) {
+	commands := []Command{}
+	files := Filesystem{}
+
+	if sys.NTP == nil {
+		return commands, files, nil
+	}
+
+	sys.AddPackage("ntp", &DEBIAN_LIKE)
+
+	configFileContent := createNTPConfigFileContent(sys.NTP)
+	files["/tmp/konfigadm-ntp.conf"] = File{Content: configFileContent}
+	sys.AddPostCommand("mv /tmp/konfigadm-ntp.conf /etc/ntp.conf", &DEBIAN_LIKE)
+	sys.AddPostCommand("systemctl restart ntp", &DEBIAN_LIKE)
+
+	return commands, files, nil
+}
+
+func createNTPConfigFileContent(ntpServers []string) string {
+	var configFileContent strings.Builder
+	for _, ntpServer := range ntpServers {
+		configFileContent.WriteString(fmt.Sprintf("server %s\n", ntpServer))
+	}
+	return configFileContent.String()
+}
+
+func (n ntp) Verify(sys *Config, results *VerifyResults, flags ...Flag) bool {
+	isNtpServiceRunning := n.serviceVerifier.VerifyService("ntp", results)
```

Comment:
Why can't we just call VerifyService? what is the valid of adding in the intermediate serviceVerifier?

Reply:
I tried to make the code testable by introducing an interface - so that a mock implementation can be passed to the test

Reply:
removed this! 👍